versions/3.1.0.md

```diff
@@ -1433,14 +1433,11 @@ application/json:
 
 ##### Considerations for File Uploads
 
-In contrast with the 2.0 specification, `file` input/output content in OpenAPI is described with the same semantics as any other schema type.  In contrast with the 3.0 specification, such schemas use the `contentEncoding` JSON Schema keyword rather than the `format` keyword.  This keyword supports all encodings defined in [RFC4648](https://tools.ietf.org/html/rfc4648), including "base64" and "base64url", as well as "quoted-printable" from [RFC2045](https://tools.ietf.org/html/rfc2045#section-6.7).
-
-JSON Schema also offers a `contentMediaType` keyword.  However, when the media type is already specified by the
-Media Type Object's key, or by the `contentType` field of an [Encoding Object](#encodingObject), the `contentMediaType` keyword SHALL be ignored if present.
+In contrast with the 2.0 specification, `file` input/output content in OpenAPI is described with the same semantics as any other schema type.  In contrast with the 3.0 specification, such schemas either omit the `type` (in place of `format: binary`), or use `contentMediaType` and `contentEncoding` with `type: string`. The `contentEncoding` keyword supports all encodings defined in [RFC4648](https://tools.ietf.org/html/rfc4648), including "base64" (which replaces `format: byte`) and "base64url", as well as "quoted-printable" from [RFC2045](https://tools.ietf.org/html/rfc2045#section-6.7).
```

Comment:
Here is my proposed rewording to clarify the priority of `Content-Encoding` headers (in various places) and the JSON Schema `contentEncoding`:
```suggestion
In contrast with the 2.0 specification, `file` input/output content in OpenAPI is described with the same semantics as any other schema type. 

In contrast with the 3.0 specification, the `format` keyword has no effect on the content-encoding of the schema. The Content-Encoding for the schema can be specified explicitly with the `Content-Encoding` parameter of the operation, for schemas in request bodies, or in a `Content-Encoding` header in the `headers` of the response, for response schemas, or in a `Content-Encoding` header in the `headers` of an Encoding object associated with a request or response body property. JSON Schema also offers a `contentEncoding` keyword, which may be used to specify the `Content-Encoding` for the schema, but `contentEncoding` SHALL be ignored if a `Content-Encoding` header is defined. The `contentEncoding` keyword supports all encodings defined in [RFC4648](https://tools.ietf.org/html/rfc4648), including "base64" (which replaces `format: byte`) and "base64url", as well as "quoted-printable" from [RFC2045](https://tools.ietf.org/html/rfc2045#section-6.7). When not explicitly specified, the content-encoding

JSON Schema also offers a `contentMediaType` keyword.  However, when the media type is already specified by the Media Type Object's key, or by the `contentType` field of an [Encoding Object](#encodingObject), the `contentMediaType` keyword SHALL be ignored if present.
```

Reply:
@mkistler the `Content-Encoding` HTTP header and `contentEncoding` JSON Schema keyword are unrelated.  The HTTP header is for things like gzip that the HTTP client and server will encode/decode automatically.  The JSON Schema keyword is for when both HTTP and the media type think that something is text, but the application needs to be informed that some specific piece of text needs to be encoded/decoded in a particular way.

Reply:
You could in fact have a gzipped payload that includes a base64-encoded PNG file.  That would use both the header and the keyword.

Reply:
@handrews I struggling to understand this all.  Do you mean to say that `contentEncoding` keyword has no effect whatsoever on the value that should/will be passed in the `Content-Encoding` header? If that is the case then it certainly will confuse some readers (like me :smile:).

In the example you give, how would this look in the API doc?  Maybe:
```
  /v1/photos:
    post:
      parameters:
        - name: Content-Encoding
           in: header
           schema:
             type: string
             enum:
               - gzip
        requestBody:
          content:
            image/png:
              schema:
                contentEncoding: base64
```

And the http request would specify:
- `Content-type: image/png`
- `Content-encoding: gzip`
- a gzipped, base64-encoded image/png

Is that right?

If I've got this right, then I think you are saying that _both_ the encoding specified in the `Content-Encoding` header _and_ the encoding specified in the `contentEncoding` keyword are applied to the request body (or whichever element they pertain to).

If that's right, I'll make a stab at rewording my suggestion to describe this.

Reply:
@mkistler yeah this is the challenge of combining multiple standards that were made in different contexts at different times.

If you go back to MIME, there is a `Content-Transfer-Encoding` header, which is where values such as `base64` from RFC 4648 (or its predecessors) are used.  Perhaps we should have used `mediaEncoding` or something like that in JSON Schema (the naming of that keyword has a slightly convoluted history that partially predates me), but we didn't.  I might have only learned about HTTP `Content-Encoding` while researching this PR, come to think of it.

I believe that what you have for the HTTP request is correct.  At this stage with the modifications y'all have made recently, I'm a little unclear on where we ended up on `type` being absent vs `"type": "string"`, but what you have (with `type` absent) is what I would have written.  And is compatible with the latest JSON Schema draft which notes that an application MAY apply a JSON Schema to other media types, including binary media types, and notes that many keywords have no sensible meaning with binary types.